xls/modules/dbe/docs/README.md

```diff
@@ -0,0 +1,289 @@
+# Dictionary coder implementation in DSLX
```

Comment:
Couple of generic comments
1. Given the long FSM backedge -- what's the expected throughput of this implementation?
2. Is the throughput/area tunable?
3. Perhaps a multi-proc implementation would be preferred to improve this.

Reply:
1. As this is an example code, we haven't really considered the throughput while writing this code. However, as far as I remember, when I was simulating the generated Verilog code, I got something in the order of 10~12 clock cycles for a single byte (with 1-cycle single-port RAMs). Even a dual-port RAM will make it somewhat better, the number is still not good enough for a practical implementation that should be targeting something like 1 byte per cycle on average (at most).
2. Not directly. We can tune the area/compression ratio by changing the hash table size, and also using single-port -vs- double-port RAM or RAMs with different latency will result in different throughput/area ratios, but there is no parameter to control this directly in a smooth way (and tbh I can't think of any way this can be done in a scalar LZ4).
3. Probably yes. I don't see how it can be improved without splitting the design.